frontend/packages/shared/src/components/TreeView/TreeViewItem/TreeViewItem.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { ElementType, KeyboardEvent, ReactNode, useEffect, useRef, useState } from 'react';
+import { useTreeViewRootContext } from '../hooks/useTreeViewRootContext';
+import { useTreeViewItemContext } from '../hooks/useTreeViewItemContext';
+import {
+  findFirstChildId,
+  findFirstNodeId,
+  findLastVisibleNodeId,
+  findNextVisibleNodeId,
+  findParentId,
+  findPreviousVisibleNodeId,
+  makeDomGroupId,
+  makeDomTreeitemId,
+} from '../utils/domUtils';
+import { AnimateHeight } from 'app-shared/components/AnimateHeight';
+import { TreeViewItemContext } from './TreeViewItemContext';
+import { ChevronDownIcon, ChevronRightIcon } from '@navikt/aksel-icons';
+import { Button } from '@digdir/design-system-react';
+import classes from './TreeViewItem.module.css';
+
+export interface TreeViewItemProps {
+  as?: ElementType;
+  children?: ReactNode;
+  label: ReactNode;
+  labelWrapper?: (children: ReactNode) => ReactNode;
+  nodeId: string;
+}
+
+export const TreeViewItem = ({
+  as = 'li',
+  children,
+  label,
+  labelWrapper = (lab) => lab,
+  nodeId,
+}: TreeViewItemProps) => {
+  const [open, setOpen] = useState(false);
+  const { selectedId, setSelectedId, rootId, focusedId, setFocusedId, focusableId } =
+    useTreeViewRootContext();
+  const { level } = useTreeViewItemContext();
+  const treeItemRef = useRef<HTMLButtonElement>(null);
+
+  useEffect(() => {
+    if (focusedId === nodeId) {
+      treeItemRef.current?.focus();
+    }
+  }, [focusedId, nodeId]);
+
+  const selected = selectedId === nodeId;
+  const focusable = focusableId === nodeId;
+
+  const handleClick = () => {
+    setOpen(!open);
```

Comment:
Ser ikke ut som det er et direkte problem i koden din, men tror jeg vil anbefale å hente ut forrige verdi av state med en funksjon som vist under. Det står litt mer om det her: https://react.dev/reference/react/useState#setstate

```suggestion
    setOpen((prevOpen) => !prevOpen);
```

Reply:
Fikset